File: src/main/java/frc/robot/subsystems/drivetrain/Gyro.java
package frc.robot.subsystems.drivetrain;

import edu.wpi.first.math.geometry.Rotation2d;

public interface Gyro {
    boolean isConnected();

    Rotation2d getAngle();

    void setGyroRotation(Rotation2d rotation);

    void update();

    void reset();

    Rotation2d getRate();
}

File: src/main/java/frc/robot/subsystems/drivetrain/Gyro.kt
package frc.robot.subsystems.drivetrain

import edu.wpi.first.math.geometry.Rotation2d
import edu.wpi.first.math.geometry.Rotation3d

interface Gyro {
    val connected: Boolean
    fun setGyroRotation(rotation: Rotation2d)
    fun update()
    fun reset()
    val angle: Rotation2d
    val rate: Rotation2d
    val rotation3d: Rotation3d
}